serialize/src/lib.rs

```diff
@@ -154,19 +151,14 @@ macro_rules! impl_uint {
 
             #[inline]
             fn serialized_size(&self) -> usize {
-                Self::SERIALIZED_SIZE
+                core::mem::size_of::<$ty>()
             }
         }
 
-        impl ConstantSerializedSize for $ty {
-            const SERIALIZED_SIZE: usize = core::mem::size_of::<$ty>();
-            const UNCOMPRESSED_SIZE: usize = Self::SERIALIZED_SIZE;
-        }
-
         impl CanonicalDeserialize for $ty {
             #[inline]
             fn deserialize<R: Read>(mut reader: R) -> Result<Self, SerializationError> {
-                let mut bytes = [0u8; Self::SERIALIZED_SIZE];
+                let mut bytes = [0u8; core::mem::size_of::<$ty>()];
```

Comment:
Is using `core::mem` safe? I thought this is the size of the in-memory representation, which could differ between architectures?

Reply:
Only `usize` differs between architectures, but for `usize` we explicitly serialize it as `u64`.

Reply:
That makes sense